src/main/java/no/nav/veilarbdirigent/config/ClientConfig.java

```diff
@@ -17,104 +13,67 @@
     }
 
     @Bean
-    public VeilarbmalverkClient veilarbmalverkClient(SystemUserTokenProvider tokenProvider) {
+    public VeilarbmalverkClient veilarbmalverkClient() {
         String url = isDevelopment().orElse(false)
                 ? createAppAdeoPreprodIngressUrl("veilarbmalverk", getEnvironment())
                 : createAppAdeoProdIngressUrl("veilarbmalverk");
-
-        return new VeilarbmalverkClientImpl(url, tokenProvider::getSystemUserToken);
+        return new VeilarbmalverkClientImpl(url);
     }
 
     @Bean
-    public VeilarbregistreringClient veilarbregistreringClient(ServiceToServiceTokenProvider serviceToServiceTokenProvider) {
+    public VeilarbregistreringClient veilarbregistreringClient(AzureAdMachineToMachineTokenClient tokenClient) {
+        var appName = "veilarbregistrering";
         String url = isDevelopment().orElse(false)
-                ? joinPaths(createDevInternalIngressUrl("veilarbregistrering"), "veilarbregistrering")
-                : createAppAdeoProdIngressUrl("veilarbregistrering");
-        String cluster = isDevelopment().orElse(false) ? "dev-gcp" : "prod-fss";
-
+                ? joinPaths(createDevInternalIngressUrl(appName), appName)
+                : createAppAdeoProdIngressUrl(appName);
+        var cluster = isDevelopment().orElse(false) ? "dev-gcp" : "prod-fss";
         return new VeilarbregistreringClientImpl(
                 url,
-                () -> serviceToServiceTokenProvider.getServiceToken(
-                        "veilarbregistrering",
-                        "paw",
-                        cluster
-                )
+                () -> tokenClient.createMachineToMachineToken(scope("veilarbregistrering", "paw", cluster))
         );
     }
 
-    @Bean
-    public ServiceToServiceTokenProvider serviceToServiceTokenProvider() {
-        return AzureAdServiceTokenProviderBuilder.builder()
-                .withEnvironmentDefaults()
-                .build();
-    }
-
-    @Bean
-    public OkHttpClient okHttpClient(SystemUserTokenProvider tokenProvider) {
-        var builder = RestClient.baseClientBuilder();
-        builder.addInterceptor(new SystemUserOidcTokenProviderInterceptor(tokenProvider));
-        return builder.build();
-    }
-
-    private static class SystemUserOidcTokenProviderInterceptor implements Interceptor {
-        private SystemUserTokenProvider systemUserTokenProvider;
-
-        private SystemUserOidcTokenProviderInterceptor(SystemUserTokenProvider systemUserTokenProvider) {
-            this.systemUserTokenProvider = systemUserTokenProvider;
-        }
-
-        @Override
-        public Response intercept(Chain chain) throws IOException {
-            Request original = chain.request();
-            Request newReq = original.newBuilder()
-                    .addHeader("Authorization", "Bearer " + systemUserTokenProvider.getSystemUserToken())
```

Comment:
jeg tror at dere trenger header verdi for AzureAD også

Reply:
Clientene legger til Auth-header selv, denne interceptoren var kun brukt av OkHttpClient men den var ikke brukt noen plass som vi kunne se